packages/design-system/src/components/Badge/button/BadgeButton.tsx

```diff
@@ -19,14 +20,19 @@ interface BadgeButtonProps {
 	 * (optional) Button click handler.
 	 */
 	onClick?: () => void;
-}
+} & Partial<DataAttributes>;
 
 const BadgeButton = forwardRef(
 	({ componentId, children, onClick, ...rest }: BadgeButtonProps, ref: Ref<HTMLButtonElement>) => {
+		const defaultTestId = 'badge-button';
+
 		return (
 			<Clickable
 				className={classnames(styles.badge__button)}
-				data-testid={componentId}
+				data-testid={
```

Comment:
I would prefer to put it in a const and reuse it

Reply:
as people want to distinct data-test and data-testid, will not be a const, will be 2 distinct values